.storybook/stories/Input/Input.tsx

```diff
@@ -18,24 +28,65 @@ export const Input: FC<InputProps> = ({
   placeholder = 'Placeholder',
   variant = 'default',
   disabled = false,
+  autoCapitalize = 'sentences',
   centerText = false,
   onChange,
+  onBlur,
+  min = 0,
+  max = 100,
 }) => {
   const [type, setType] = useState<InputProps['variant'] | 'active'>(variant);
-  const onFocus = useCallback(() => setType('active'), []);
-  const onBlur = useCallback(() => setType(variant), [variant]);
+
+  const onChangeInput = useCallback((value: string) => {
+    const newValue = keyboardType === 'default' ? value : getNumberValue(value);
+    onChange(newValue);
+  }, []);
+
+  const onFocusInput = useCallback(
+    (value: NativeSyntheticEvent<TextInputFocusEventData>) => {
+      setType('active');
+    },
+    [],
+  );
+
+  const onBlurInput = useCallback(
+    (value: NativeSyntheticEvent<TextInputEndEditingEventData>) => {
+      setType(variant);
+      onBlur?.(value.nativeEvent.text);
+    },
+    [variant],
+  );
+
+  const getNumberValue = (value: string): string => {
+    if (value === '') {
+      value = min.toString();
+    }
+    const parts = value.replace(',', '.').split('.').slice(0, 2);
+    parts[0] = parseInt(parts[0]).toString();
+    const newValue = parts.join('.');
+    const newValueNumeric = parseFloat(newValue);
+    const finalValue =
+      newValueNumeric < min
+        ? min.toString()
+        : newValueNumeric > max
+        ? max.toString()
+        : newValue;
```

Comment:
Don't use nested ternary operators. Just write
```
if (newValueNumeric < min) {
return min.toString()
} else if (newValueNumeric > max) {
return max.toString()
}
return newValueNumeric
```
This takes the same number of lines, but it's much more readable and easy to understand.